### PR TITLE
Project Auth Token Step

### PR DESCRIPTION
**Pull request by beamhafsa**

### CODE REVIEW

**Comment by husseinbani** [comment]
I think you should merge it to Refactoring branch as it has the latest changes and it still not merged, @MJaghoub please confirm?

**Comment by husseinbani** [comment]
Fix the conflict

**Comment by beamhafsa** [comment]
Conflicts have been fixed

File: beam-automation-rest/src/test/resources/application-local.yml [hunk 1]
```diff
@@ -1,3 +1,13 @@
 app:
     lmsman:
-        url: http://aef8dbbd6294a11e9a2a4021e775f7f0-2112399806.eu-west-1.elb.amazonaws.com/lms/
\ No newline at end of file
+        url: http://aef8dbbd6294a11e9a2a4021e775f7f0-2112399806.eu-west-1.elb.amazonaws.com/lms/
+    auth0:
```
Comment (MJaghoub) [inline beam-automation-rest/src/test/resources/application-local.yml]: Please add those parameters to `server` property file

Comment (beamhafsa) [reply beam-automation-rest/src/test/resources/application-local.yml]: Added

File: beam-automation-rest/src/test/java/me/beam/automation/rest/CucumberConfigTest.java [hunk 2]
```diff
@@ -5,6 +5,9 @@
 import org.junit.runner.RunWith;
 
 @RunWith(Cucumber.class)
-@CucumberOptions(features = "src/test/resources/features", plugin = {"pretty", "html:target/cucumber"})
+@CucumberOptions(features = "src/test/resources/features",
+    plugin = {"pretty", "html:target/cucumber"},
+    glue = {"me.beam.automation.rest.stepdefs"},
+    tags = {"@skip_scenario"})
```
Comment (MJaghoub) [inline beam-automation-rest/src/test/java/me/beam/automation/rest/CucumberConfigTest.java]: why did you add this tag here?

File: beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/LmsStepsImpl.java [hunk 3]
```diff
@@ -75,21 +102,26 @@ public void memberIdShouldBeSameAsTheRequestMemberId(String memberId) {
...
-			throw new RestAutomationException("Error in converting dataTable to lmsRequest", e);
-		}
+	@Given("user creates an auth0 token")
+	public void userCreatesAnAuthToken() {
+		Auth0Token auth0Token = new Auth0Token(domain, clientId, grantType, username, password,
+				clientSecret, audience, realm);
+		HttpHeaders headers = new HttpHeaders();
+		headers.setContentType(MediaType.APPLICATION_JSON);
+		Map<String, Object> response = call(domain, auth0Token.getAuth0Body(), HttpMethod.POST, headers);
+		this.token = (String) response.get("access_token");
```
Comment (MJaghoub) [inline beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/LmsStepsImpl.java]: Please move `access_token ` to Constant

File: beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/dto/LmsRequest.java [hunk 4]
```diff
@@ -18,4 +18,34 @@
 	private Integer totalAmountSpent;
 	private Integer sponsorId;
 	private Integer programId;
+
+	public LmsRequest(String memberId, String location, String currency, String serviceProvider,
```
Comment (MJaghoub) [inline beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/dto/LmsRequest.java]: you can use `@AllArgsConstructor` from `lombok` and remove this constructor

File: beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/transformer/LmsRequestTransformer.java [hunk 5]
```diff
@@ -1,40 +1,45 @@
...
+	public Locale locale() {
+		return Locale.ENGLISH;
+	}
+
+	public void configureTypeRegistry(TypeRegistry typeRegistry) {
+		typeRegistry.defineDataTableType(new DataTableType(LmsRequest.class,
+                (Map<String, String> row) -> {
+                    String memberId = row.get("memberId");
+                    String currency = row.get("currency");
+                    Integer totalAmountSpent = Integer.parseInt(row.get("totalAmountSpent"));
```
Comment (MJaghoub) [inline beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/transformer/LmsRequestTransformer.java]: please use Constant for all parameters

File: beam-automation-rest/src/test/java/me/beam/automation/rest/CucumberConfigTest.java (hunk 2 above)
Comment (beamhafsa) [reply beam-automation-rest/src/test/java/me/beam/automation/rest/CucumberConfigTest.java]: Added for testing few test cases. Removed.

File: beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/LmsStepsImpl.java (hunk 3 above)
Comment (beamhafsa) [reply beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/LmsStepsImpl.java]: Added in constants

File: beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/dto/LmsRequest.java (hunk 4 above)
Comment (beamhafsa) [reply beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/dto/LmsRequest.java]: Added  @AllArgsConstructor but for intellij idea we have to add a plugin to support Lombok.

File: beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/transformer/LmsRequestTransformer.java (hunk 5 above)
Comment (beamhafsa) [reply beam-automation-rest/src/test/java/me/beam/automation/rest/stepdefs/transformer/LmsRequestTransformer.java]: Added and used constants
